melodies_monet/driver.py

```diff
@@ -661,7 +662,29 @@ def open_models(self, time_interval=None):
                         m.scrip_file = self.control_dict['model'][mod]['scrip_file']
                     else:
                         raise ValueError( '"Scrip_file" must be provided for unstructured grid output!' )
-                        
+
+                # maybe set projection
+                proj_in = self.control_dict['model'][mod].get("projection")
+                if proj_in == "None":
+                    print(
+                        f"NOTE: model.{mod}.projection is {proj_in!r} (str), "
+                        "but we assume you want `None` (Python null sentinel). "
+                        "To avoid this warning, "
+                        "update your control file to remove the projection setting "
+                        "or set to `~` or `null` if you want null value in YAML."
+                    )
+                    proj_in = None
+                if proj_in is not None:
+                    if isinstance(proj_in, str) and proj_in.startswith("model:"):
+                        m.proj = proj_in
+                    else:
+                        import cartopy.crs as ccrs
+
+                        if isinstance(proj_in, ccrs.Projection):
+                            m.proj = proj_in
+                        else:
+                            m.proj = ccrs.Projection(proj_in)
```

Comment:
I may be missing something but I can't get this to work when I've set `model.<model>.projection` with proj4 parameters. Is this an error on my part in how `model.<model>.projection` is set or an issue with using `ccrs.Projection` in this way?

The docs for ccrs.Projection say it needs to be an iterable of key-value pairs containing proj4 parameters, so I'll specify it as a dict and use the parameters from `ccrs.PlateCarree().to_dict()`. In the control file I then have `projection: {'proj': 'eqc', 'lat_ts': 0, 'lat_0': 0, 'lon_0': 0, 'x_0': 0, 'y_0': 0, 'ellps': 'WGS84', 'to_meter': 111319.490793274, 'no_defs': None, 'type': 'crs'}`.

Running a spatial_overlay plot I get a NotImplementedError from `boundary` in `cartopy.crs` with the following traceback:

> Traceback (most recent call last):
  File "/wrk/charkins/melodies_monet/NOAA_CSL_main/test/test_filter_obs.py", line 18, in <module>
    an.plotting()
  File "/wrk/charkins/melodies_monet/NOAA_CSL_main/MELODIES-MONET/melodies_monet/driver.py", line 1277, in plotting
    splots.make_spatial_overlay(
  File "/wrk/charkins/melodies_monet/NOAA_CSL_main/MELODIES-MONET/melodies_monet/plots/surfplots.py", line 751, in make_spatial_overlay
    ax = vmodel_mean.monet.quick_contourf(cbar_kwargs=cbar_kwargs, figsize=map_kwargs['figsize'], map_kws=map_kwargs,
  File "/wrk/charkins/melodies_monet/NOAA_CSL_main/monet/monet/monet_accessor.py", line 1192, in quick_contourf
    ax = draw_map(**map_kws)
  File "/wrk/charkins/melodies_monet/NOAA_CSL_main/monet/monet/plots/mapgen.py", line 53, in draw_map
    f, ax = plt.subplots(figsize=figsize, subplot_kw={"projection": crs})
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/pyplot.py", line 1443, in subplots
    axs = fig.subplots(nrows=nrows, ncols=ncols, sharex=sharex, sharey=sharey,
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/figure.py", line 894, in subplots
    axs = gs.subplots(sharex=sharex, sharey=sharey, squeeze=squeeze,
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/gridspec.py", line 308, in subplots
    axarr[row, col] = figure.add_subplot(
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/figure.py", line 745, in add_subplot
    ax = subplot_class_factory(projection_class)(self, *args, **pkw)
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/axes/_subplots.py", line 34, in __init__
    self._axes_class.__init__(self, fig, [0, 0, 1, 1], **kwargs)
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 419, in __init__
    super().__init__(*args, **kwargs)
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 653, in __init__
    self.clear()
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 599, in clear
    self.__clear()
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 586, in __clear
    self._boundary()
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 1552, in _boundary
    path, = cpatch.geos_to_path(self.projection.boundary)
  File "/home/charkins/miniconda3/envs/melodies-monet_NOAA_CSL/lib/python3.9/site-packages/cartopy/crs.py", line 682, in boundary
    raise NotImplementedError
NotImplementedError

I also tried setting `projection` with a proj4 string and got the same result.

Reply:
Yeah I also experienced that. For example, when I put the proj4 string for `ccrs.PlateCarre()`, it failed at plotting since `.boundary` doesn't get defined. Maybe this is a Cartopy limitation, as it doesn't seem possible to set the bounds it needs with that. But setting it to `'EPSG:4326'`, e.g., does work.

Reply:
OK well if there isn't much we can do about it then it's fine, I also looked into setting the bounds it needs with a proj4 string and couldn't find a way.

Reply:
Thanks. We can come back to this if we learn more.